base/errorshow.jl

```diff
@@ -98,10 +98,40 @@ function showerror(io::IO, ex, bt; backtrace=true)
     end
 end
 
+function stacktrace_path(file::Union{Nothing, String}, line::Union{Nothing, Int64})
+    realfile = if !isnothing(file) && file != "" && !startswith(String(file), "REPL")
+        fixup_stdlib_path(String(file))
+    end
+    pathstr = file
+    if !isnothing(pathstr)
+        stacktrace_expand_basepaths() && (pathstr = something(find_source_file(file), file))
+        stacktrace_contract_userdir() && (pathstr = contractuser(pathstr))
+    end
+    linestr = if !isnothing(line) && line > 0
+        string(something(pathstr, ""), ':', line)
+    else
+        something(pathstr, "")
+    end
+    if !isnothing(realfile)
+        flen = ncodeunits(basename(realfile))
+        llen = ncodeunits(string(line))
```

Comment:
Might be helpful not to recompute `string(line)` here, as this seems to be the wrong length if the `linestr` computation was nothing or <= 0

Reply:
Ah yep, I think `llen` should be folded into the `linestr` assignment. I'll push that change.